{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Exception in Tkinter callback\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\Acer\\AppData\\Local\\Programs\\Python\\Python310\\lib\\tkinter\\__init__.py\", line 1921, in __call__\n",
      "    return self.func(*args)\n",
      "  File \"C:\\Users\\Acer\\AppData\\Local\\Temp\\ipykernel_15444\\3068073414.py\", line 11, in language_translator\n",
      "    x = translator()\n",
      "TypeError: 'Translator' object is not callable\n",
      "Exception in Tkinter callback\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\Acer\\AppData\\Local\\Programs\\Python\\Python310\\lib\\tkinter\\__init__.py\", line 1921, in __call__\n",
      "    return self.func(*args)\n",
      "  File \"C:\\Users\\Acer\\AppData\\Local\\Temp\\ipykernel_15444\\3068073414.py\", line 11, in language_translator\n",
      "    x = translator()\n",
      "TypeError: 'Translator' object is not callable\n",
      "Exception in Tkinter callback\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\Acer\\AppData\\Local\\Programs\\Python\\Python310\\lib\\tkinter\\__init__.py\", line 1921, in __call__\n",
      "    return self.func(*args)\n",
      "  File \"C:\\Users\\Acer\\AppData\\Local\\Temp\\ipykernel_15444\\3068073414.py\", line 11, in language_translator\n",
      "    x = translator()\n",
      "TypeError: 'Translator' object is not callable\n",
      "Exception in Tkinter callback\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\Acer\\AppData\\Local\\Programs\\Python\\Python310\\lib\\tkinter\\__init__.py\", line 1921, in __call__\n",
      "    return self.func(*args)\n",
      "  File \"C:\\Users\\Acer\\AppData\\Local\\Temp\\ipykernel_15444\\3068073414.py\", line 11, in language_translator\n",
      "    x = translator()\n",
      "TypeError: 'Translator' object is not callable\n",
      "Exception in Tkinter callback\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\Acer\\AppData\\Local\\Programs\\Python\\Python310\\lib\\tkinter\\__init__.py\", line 1921, in __call__\n",
      "    return self.func(*args)\n",
      "  File \"C:\\Users\\Acer\\AppData\\Local\\Temp\\ipykernel_15444\\3068073414.py\", line 11, in language_translator\n",
      "    x = translator()\n",
      "TypeError: 'Translator' object is not callable\n",
      "Exception in Tkinter callback\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\Acer\\AppData\\Local\\Programs\\Python\\Python310\\lib\\tkinter\\__init__.py\", line 1921, in __call__\n",
      "    return self.func(*args)\n",
      "  File \"C:\\Users\\Acer\\AppData\\Local\\Temp\\ipykernel_15444\\3068073414.py\", line 11, in language_translator\n",
      "    x = translator()\n",
      "TypeError: 'Translator' object is not callable\n",
      "Exception in Tkinter callback\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\Acer\\AppData\\Local\\Programs\\Python\\Python310\\lib\\tkinter\\__init__.py\", line 1921, in __call__\n",
      "    return self.func(*args)\n",
      "  File \"C:\\Users\\Acer\\AppData\\Local\\Temp\\ipykernel_15444\\3068073414.py\", line 11, in language_translator\n",
      "    x = translator()\n",
      "TypeError: 'Translator' object is not callable\n",
      "Exception in Tkinter callback\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\Acer\\AppData\\Local\\Programs\\Python\\Python310\\lib\\tkinter\\__init__.py\", line 1921, in __call__\n",
      "    return self.func(*args)\n",
      "  File \"C:\\Users\\Acer\\AppData\\Local\\Temp\\ipykernel_15444\\3068073414.py\", line 11, in language_translator\n",
      "    x = translator()\n",
      "TypeError: 'Translator' object is not callable\n",
      "Exception in Tkinter callback\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\Acer\\AppData\\Local\\Programs\\Python\\Python310\\lib\\tkinter\\__init__.py\", line 1921, in __call__\n",
      "    return self.func(*args)\n",
      "  File \"C:\\Users\\Acer\\AppData\\Local\\Temp\\ipykernel_15444\\3068073414.py\", line 11, in language_translator\n",
      "    x = translator()\n",
      "TypeError: 'Translator' object is not callable\n",
      "Exception in Tkinter callback\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\Acer\\AppData\\Local\\Programs\\Python\\Python310\\lib\\tkinter\\__init__.py\", line 1921, in __call__\n",
      "    return self.func(*args)\n",
      "  File \"C:\\Users\\Acer\\AppData\\Local\\Temp\\ipykernel_15444\\3068073414.py\", line 11, in language_translator\n",
      "    x = translator()\n",
      "TypeError: 'Translator' object is not callable\n",
      "Exception in Tkinter callback\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\Acer\\AppData\\Local\\Programs\\Python\\Python310\\lib\\tkinter\\__init__.py\", line 1921, in __call__\n",
      "    return self.func(*args)\n",
      "  File \"C:\\Users\\Acer\\AppData\\Local\\Temp\\ipykernel_15444\\3068073414.py\", line 11, in language_translator\n",
      "    x = translator()\n",
      "TypeError: 'Translator' object is not callable\n",
      "Exception in Tkinter callback\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\Acer\\AppData\\Local\\Programs\\Python\\Python310\\lib\\tkinter\\__init__.py\", line 1921, in __call__\n",
      "    return self.func(*args)\n",
      "  File \"C:\\Users\\Acer\\AppData\\Local\\Temp\\ipykernel_15444\\3068073414.py\", line 11, in language_translator\n",
      "    x = translator()\n",
      "TypeError: 'Translator' object is not callable\n",
      "Exception in Tkinter callback\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\Acer\\AppData\\Local\\Programs\\Python\\Python310\\lib\\tkinter\\__init__.py\", line 1921, in __call__\n",
      "    return self.func(*args)\n",
      "  File \"C:\\Users\\Acer\\AppData\\Local\\Temp\\ipykernel_15444\\3068073414.py\", line 11, in language_translator\n",
      "    x = translator()\n",
      "TypeError: 'Translator' object is not callable\n",
      "Exception in Tkinter callback\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\Acer\\AppData\\Local\\Programs\\Python\\Python310\\lib\\tkinter\\__init__.py\", line 1921, in __call__\n",
      "    return self.func(*args)\n",
      "  File \"C:\\Users\\Acer\\AppData\\Local\\Temp\\ipykernel_15444\\3068073414.py\", line 11, in language_translator\n",
      "    x = translator()\n",
      "TypeError: 'Translator' object is not callable\n",
      "Exception in Tkinter callback\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\Acer\\AppData\\Local\\Programs\\Python\\Python310\\lib\\tkinter\\__init__.py\", line 1921, in __call__\n",
      "    return self.func(*args)\n",
      "  File \"C:\\Users\\Acer\\AppData\\Local\\Temp\\ipykernel_15444\\3068073414.py\", line 11, in language_translator\n",
      "    x = translator()\n",
      "TypeError: 'Translator' object is not callable\n",
      "Exception in Tkinter callback\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\Acer\\AppData\\Local\\Programs\\Python\\Python310\\lib\\tkinter\\__init__.py\", line 1921, in __call__\n",
      "    return self.func(*args)\n",
      "  File \"C:\\Users\\Acer\\AppData\\Local\\Temp\\ipykernel_15444\\3068073414.py\", line 11, in language_translator\n",
      "    x = translator()\n",
      "TypeError: 'Translator' object is not callable\n",
      "Exception in Tkinter callback\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\Acer\\AppData\\Local\\Programs\\Python\\Python310\\lib\\tkinter\\__init__.py\", line 1921, in __call__\n",
      "    return self.func(*args)\n",
      "  File \"C:\\Users\\Acer\\AppData\\Local\\Temp\\ipykernel_15444\\3068073414.py\", line 11, in language_translator\n",
      "    x = translator()\n",
      "TypeError: 'Translator' object is not callable\n",
      "Exception in Tkinter callback\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\Acer\\AppData\\Local\\Programs\\Python\\Python310\\lib\\tkinter\\__init__.py\", line 1921, in __call__\n",
      "    return self.func(*args)\n",
      "  File \"C:\\Users\\Acer\\AppData\\Local\\Temp\\ipykernel_15444\\3068073414.py\", line 11, in language_translator\n",
      "    x = translator()\n",
      "TypeError: 'Translator' object is not callable\n",
      "Exception in Tkinter callback\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\Acer\\AppData\\Local\\Programs\\Python\\Python310\\lib\\tkinter\\__init__.py\", line 1921, in __call__\n",
      "    return self.func(*args)\n",
      "  File \"C:\\Users\\Acer\\AppData\\Local\\Temp\\ipykernel_15444\\3068073414.py\", line 11, in language_translator\n",
      "    x = translator()\n",
      "TypeError: 'Translator' object is not callable\n",
      "Exception in Tkinter callback\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\Acer\\AppData\\Local\\Programs\\Python\\Python310\\lib\\tkinter\\__init__.py\", line 1921, in __call__\n",
      "    return self.func(*args)\n",
      "  File \"C:\\Users\\Acer\\AppData\\Local\\Temp\\ipykernel_15444\\3068073414.py\", line 11, in language_translator\n",
      "    x = translator()\n",
      "TypeError: 'Translator' object is not callable\n",
      "Exception in Tkinter callback\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\Acer\\AppData\\Local\\Programs\\Python\\Python310\\lib\\tkinter\\__init__.py\", line 1921, in __call__\n",
      "    return self.func(*args)\n",
      "  File \"C:\\Users\\Acer\\AppData\\Local\\Temp\\ipykernel_15444\\3068073414.py\", line 11, in language_translator\n",
      "    x = translator()\n",
      "TypeError: 'Translator' object is not callable\n",
      "Exception in Tkinter callback\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\Acer\\AppData\\Local\\Programs\\Python\\Python310\\lib\\tkinter\\__init__.py\", line 1921, in __call__\n",
      "    return self.func(*args)\n",
      "  File \"C:\\Users\\Acer\\AppData\\Local\\Temp\\ipykernel_15444\\3068073414.py\", line 11, in language_translator\n",
      "    x = translator()\n",
      "TypeError: 'Translator' object is not callable\n",
      "Exception in Tkinter callback\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\Acer\\AppData\\Local\\Programs\\Python\\Python310\\lib\\tkinter\\__init__.py\", line 1921, in __call__\n",
      "    return self.func(*args)\n",
      "  File \"C:\\Users\\Acer\\AppData\\Local\\Temp\\ipykernel_15444\\3068073414.py\", line 11, in language_translator\n",
      "    x = translator()\n",
      "TypeError: 'Translator' object is not callable\n",
      "Exception in Tkinter callback\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\Acer\\AppData\\Local\\Programs\\Python\\Python310\\lib\\tkinter\\__init__.py\", line 1921, in __call__\n",
      "    return self.func(*args)\n",
      "  File \"C:\\Users\\Acer\\AppData\\Local\\Temp\\ipykernel_15444\\3068073414.py\", line 11, in language_translator\n",
      "    x = translator()\n",
      "TypeError: 'Translator' object is not callable\n"
     ]
    }
   ],
   "source": [
    "from tkinter import *\n",
    "from tkinter import ttk\n",
    "from googletrans import Translator,LANGUAGES\n",
    "\n",
    "translator=Translator()\n",
    "window=Tk()\n",
    "window.title(\"Language Translator\")\n",
    "window.geometry(\"1000x600\")\n",
    "window.config(bg=\"#2929a3\")\n",
    "def language_translator():\n",
    "    x = translator()\n",
    "    result=x.translate(text=Input.get(1.0,END),src=src_lang.get(),dest=dest_lang.get())\n",
    "    output.delete(1.0,END)\n",
    "    output.insert(END,result.text)\n",
    "Label(window,text=\"Welcome to Language Translator App\",height=3,width=50,font=\"arial 25 bold\",bg=\"black\",foreground=\"white\").pack()\n",
    "Label(window,text=\"Source Language\",font=\"arial 15 bold\",foreground=\"black\").place(x=20,y=150)\n",
    "Label(window,text=\"Output Language\",font=\"arial 15 bold\",foreground=\"black\").place(x=750,y=150)\n",
    "Input=Text(window,font=\"arial 15 bold\",height=12,width=60)\n",
    "Input.place(x=10,y=200)\n",
    "output=Text(window,font=\"arial 15 bold\",height=12,width=60)\n",
    "output.place(x=750,y=200)\n",
    "lang_list=list(LANGUAGES.values())\n",
    "src_lang=ttk.Combobox(window,values=lang_list)\n",
    "src_lang.place(x=200,y=150)\n",
    "dest_lang=ttk.Combobox(window,values=lang_list)\n",
    "dest_lang.place(x=950,y=150)\n",
    "Button(window,text=\"Translate\",font=\"arial 35 bold\",command=language_translator).place(x=600,y=550)\n",
    "window.mainloop()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "#tkinter library used for to create graphical user interface\n",
    "import tkinter #tkinter library used for to create graphical user interface\n",
    "from tkinter import *\n",
    "from tkinter import ttk\n",
    "from googletrans import Translator,LANGUAGES\n",
    "#Translator is a class which is used to translate the text from one language to another language\n",
    "#LANGUAGES is a dictionary which contains all the languages in the world\n",
    "\n",
    "\n",
    "translator=Translator()\n",
    "window=Tk()\n",
    "window.title(\"Language Translator\")\n",
    "window.geometry(\"1000x600\")\n",
    "window.config(bg=\"#2929a3\")\n",
    "def lang1():\n",
    "    global words\n",
    "    x=Translator()\n",
    "    result=x.translate(text=Input.get(1.0,END),src=src_lang.get(),dest=des_lang.get())\n",
    "    output.delete(1.0,END)\n",
    "    output.insert(END,result.text)\n",
    "Label(window,text=\"Welcome to Language Translator App\",height=3,width=50,font=\"arial 25 bold\",bg=\"black\",foreground=\"white\").pack()\n",
    "Label(window,text=\"Source Language\",font=\"arial 15 bold\",foreground=\"black\").place(x=20,y=150)\n",
    "Label(window,text=\"Output Language\",font=\"arial 15 bold\",foreground=\"black\").place(x=750,y=150)\n",
    "Input=Text(window,font=\"arial 15 bold\",height=12,width=60)\n",
    "Input.place(x=10,y=200)\n",
    "output=Text(window,font=\"arial 15 bold\",height=12,width=60)\n",
    "output.place(x=750,y=200)\n",
    "src_lan=list(LANGUAGES.values())\n",
    "des_lan=list(LANGUAGES.values())\n",
    "src_lang=ttk.Combobox(window,values=src_lan)\n",
    "src_lang.place(x=200,y=150)\n",
    "des_lang=ttk.Combobox(window,values=des_lan)\n",
    "des_lang.place(x=950,y=150)\n",
    "Button(window,text=\"Translate\",font=\"arial 35 bold\",command=lang1).place(x=600,y=550)\n",
    "window.mainloop()\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Collecting SpeechRecognition\n",
      "  Downloading SpeechRecognition-3.10.3-py2.py3-none-any.whl (32.8 MB)\n",
      "     ---------------------------------------- 32.8/32.8 MB 5.5 MB/s eta 0:00:00\n",
      "Requirement already satisfied: requests>=2.26.0 in c:\\users\\acer\\appdata\\local\\programs\\python\\python310\\lib\\site-packages (from SpeechRecognition) (2.31.0)\n",
      "Collecting typing-extensions\n",
      "  Downloading typing_extensions-4.11.0-py3-none-any.whl (34 kB)\n",
      "Requirement already satisfied: urllib3<3,>=1.21.1 in c:\\users\\acer\\appdata\\local\\programs\\python\\python310\\lib\\site-packages (from requests>=2.26.0->SpeechRecognition) (2.2.1)\n",
      "Requirement already satisfied: idna<4,>=2.5 in c:\\users\\acer\\appdata\\local\\programs\\python\\python310\\lib\\site-packages (from requests>=2.26.0->SpeechRecognition) (2.10)\n",
      "Requirement already satisfied: certifi>=2017.4.17 in c:\\users\\acer\\appdata\\local\\programs\\python\\python310\\lib\\site-packages (from requests>=2.26.0->SpeechRecognition) (2024.2.2)\n",
      "Requirement already satisfied: charset-normalizer<4,>=2 in c:\\users\\acer\\appdata\\local\\programs\\python\\python310\\lib\\site-packages (from requests>=2.26.0->SpeechRecognition) (3.3.2)\n",
      "Installing collected packages: typing-extensions, SpeechRecognition\n",
      "Successfully installed SpeechRecognition-3.10.3 typing-extensions-4.11.0\n",
      "Note: you may need to restart the kernel to use updated packages.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "WARNING: You are using pip version 22.0.4; however, version 24.0 is available.\n",
      "You should consider upgrading via the 'c:\\Users\\Acer\\AppData\\Local\\Programs\\Python\\Python310\\python.exe -m pip install --upgrade pip' command.\n"
     ]
    }
   ],
   "source": [
    "pip install SpeechRecognition"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Listening...\n",
      "Listening...\n",
      "Google Speech Recognition could not understand audio\n",
      "Listening...\n",
      "Listening...\n"
     ]
    }
   ],
   "source": [
    "import tkinter\n",
    "from tkinter import *\n",
    "from tkinter import ttk\n",
    "from googletrans import Translator, LANGUAGES\n",
    "import speech_recognition as sr\n",
    "\n",
    "translator = Translator()\n",
    "window = Tk()\n",
    "window.title(\"Language Translator\")\n",
    "window.geometry(\"1000x600\")\n",
    "window.config(bg=\"#2929a3\")\n",
    "\n",
    "def lang1():\n",
    "    global words\n",
    "    x = Translator()\n",
    "    result = x.translate(text=Input.get(1.0, END), src=src_lang.get(), dest=des_lang.get())\n",
    "    output.delete(1.0, END)\n",
    "    output.insert(END, result.text)\n",
    "\n",
    "def recognize_speech():\n",
    "    recognizer = sr.Recognizer()\n",
    "    with sr.Microphone() as source:\n",
    "        print(\"Listening...\")\n",
    "        recognizer.adjust_for_ambient_noise(source, duration=1)\n",
    "        audio = recognizer.listen(source)\n",
    "\n",
    "    try:\n",
    "        recognized_text = recognizer.recognize_google(audio)\n",
    "        Input.delete(1.0, END)\n",
    "        Input.insert(END, recognized_text)\n",
    "    except sr.UnknownValueError:\n",
    "        print(\"Google Speech Recognition could not understand audio\")\n",
    "    except sr.RequestError as e:\n",
    "        print(f\"Could not request results from Google Speech Recognition service; {e}\")\n",
    "\n",
    "Label(window, text=\"Welcome to Language Translator App\", height=3, width=50, font=\"arial 25 bold\", bg=\"black\",\n",
    "      foreground=\"white\").pack()\n",
    "Label(window, text=\"Source Language\", font=\"arial 15 bold\", foreground=\"black\").place(x=20, y=150)\n",
    "Label(window, text=\"Output Language\", font=\"arial 15 bold\", foreground=\"black\").place(x=750, y=150)\n",
    "Input = Text(window, font=\"arial 15 bold\", height=12, width=60)\n",
    "Input.place(x=10, y=200)\n",
    "output = Text(window, font=\"arial 15 bold\", height=12, width=60)\n",
    "output.place(x=750, y=200)\n",
    "src_lan = list(LANGUAGES.values())\n",
    "des_lan = list(LANGUAGES.values())\n",
    "src_lang = ttk.Combobox(window, values=src_lan)\n",
    "src_lang.place(x=200, y=150)\n",
    "des_lang = ttk.Combobox(window, values=des_lan)\n",
    "des_lang.place(x=950, y=150)\n",
    "Button(window, text=\"Translate\", font=\"arial 35 bold\", command=lang1).place(x=600, y=550)\n",
    "Button(window, text=\"Speech to Text\", font=\"arial 20 bold\", command=recognize_speech).place(x=400, y=550)\n",
    "window.mainloop()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Collecting PyAudioNote: you may need to restart the kernel to use updated packages.\n",
      "\n",
      "  Downloading PyAudio-0.2.14-cp310-cp310-win_amd64.whl (164 kB)\n",
      "     -------------------------------------- 164.1/164.1 KB 3.3 MB/s eta 0:00:00\n",
      "Installing collected packages: PyAudio\n",
      "Successfully installed PyAudio-0.2.14\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "WARNING: You are using pip version 22.0.4; however, version 24.0 is available.\n",
      "You should consider upgrading via the 'c:\\Users\\Acer\\AppData\\Local\\Programs\\Python\\Python310\\python.exe -m pip install --upgrade pip' command.\n"
     ]
    }
   ],
   "source": [
    "pip install PyAudio"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import tkinter\n",
    "from tkinter import *\n",
    "from tkinter import ttk\n",
    "from googletrans import Translator, LANGUAGES\n",
    "import speech_recognition as sr\n",
    "\n",
    "translator = Translator()\n",
    "window = Tk()\n",
    "window.title(\"Language Translator\")\n",
    "window.geometry(\"1000x600\")\n",
    "window.config(bg=\"#2929a3\")\n",
    "\n",
    "def lang1():\n",
    "    global words\n",
    "    x = Translator()\n",
    "    result = x.translate(text=Input.get(1.0, END), src=src_lang.get(), dest=des_lang.get())\n",
    "    output.delete(1.0, END)\n",
    "    output.insert(END, result.text)\n",
    "\n",
    "def recognize_speech():\n",
    "    recognizer = sr.Recognizer()\n",
    "    with sr.Microphone() as source:\n",
    "        print(\"Listening...\")\n",
    "        recognizer.adjust_for_ambient_noise(source, duration=1)\n",
    "        audio = recognizer.listen(source)\n",
    "\n",
    "    try:\n",
    "        recognized_text = recognizer.recognize_google(audio)\n",
    "        Input.delete(1.0, END)\n",
    "        Input.insert(END, recognized_text)\n",
    "    except sr.UnknownValueError:\n",
    "        print(\"Google Speech Recognition could not understand audio\")\n",
    "    except sr.RequestError as e:\n",
    "        print(f\"Could not request results from Google Speech Recognition service; {e}\")\n",
    "\n",
    "Label(window, text=\"Welcome to Language Translator App\", height=3, width=50, font=\"arial 25 bold\", bg=\"black\",\n",
    "      foreground=\"white\").pack()\n",
    "Label(window, text=\"Source Language\", font=\"arial 15 bold\", foreground=\"black\").place(x=20, y=150)\n",
    "Label(window, text=\"Output Language\", font=\"arial 15 bold\", foreground=\"black\").place(x=750, y=150)\n",
    "Input = Text(window, font=\"arial 15 bold\", height=12, width=60)\n",
    "Input.place(x=10, y=200)\n",
    "output = Text(window, font=\"arial 15 bold\", height=12, width=60)\n",
    "output.place(x=750, y=200)\n",
    "src_lan = list(LANGUAGES.values())\n",
    "des_lan = list(LANGUAGES.values())\n",
    "src_lang = ttk.Combobox(window, values=src_lan)\n",
    "src_lang.place(x=200, y=150)\n",
    "des_lang = ttk.Combobox(window, values=des_lan)\n",
    "des_lang.place(x=950, y=150)\n",
    "Button(window, text=\"Translate\", font=\"arial 35 bold\", command=lang1).place(x=600, y=550)\n",
    "Button(window, text=\"Speech to Text\", font=\"arial 20 bold\", command=recognize_speech).place(x=400, y=550)\n",
    "window.mainloop()"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
